py-polars/polars/internals/series/datetime.py

```diff
@@ -911,7 +917,7 @@ def cast_time_unit(self, tu: TimeUnit) -> pli.Series:
 
         """
 
-    def with_time_zone(self, tz: str) -> pli.Series:
+    def convert_time_zone(self, tz: str) -> pli.Series:
```

Comment:
And unrelated and possibly out of scope for this PR: should we allow `ZoneInfo` inputs here, in addition to strings?

Reply:
Adding another accepted type doesn't change users code right?

Reply:
Correct - it will not be a breaking change.

We'll just add the type to the method signature and something like this to the body:

```python
if isinstance(tz, ZoneInfo):
    tz = str(tz)
```

Reply:
sure - `datetime.timezone` objects too I presume? pandas handles them:
```python
In [6]: Timestamp.now().tz_localize(dt.timezone(timedelta(hours=1)))
Out[6]: Timestamp('2023-02-10 10:56:16.305580+0100', tz='UTC+01:00')

In [7]: Timestamp.now().tz_localize(zoneinfo.ZoneInfo(key='US/Pacific'))
Out[7]: Timestamp('2023-02-10 10:56:42.151468-0800', tz='US/Pacific')
```

Reply:
Hadn't thought of that, but sure! That would be nice.